specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1287 @@
+Notice that ';' is used as the delimiter between each argument.
+
+To retrieve the arguments from the serialized string, developers will use 
+ArgumentSerializer::Deserialize(String argumentString), a static function which returns a mapping of the 
+all the arguments received from the action.
+
+Below is an example usage:
+```c#
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+using namespace winrt::Microsoft::Windows::AppNotifications;
+
+// Retrieve the activation args using Windows App SDK AppLifecycle functions
+AppActivationArguments args { AppLifecycle::AppInstance::GetCurrent().GetActivatedEventArgs() };
+ExtendedActivationKind kind { args.Kind() };
+
+// Check the activation kind
+if (kind == ExtendedActivationKind::AppNotification)
+{
+    AppNotificationActivatedEventArgs appActivatedArgs { args.Data().as<AppNotificationActivatedEventArgs>() };
+
+    // Retrieve the serialized arguments from AppNotificationActivatedEventArgs
+    winrt::hstring argumentString{ appActivatedArgs.Argument() };
+
+    // Deserialize the string to a mapping of the arguments
+    Windows.Foundation.Collections::IMap<winrt::hstring, winrt::hstring> argumentMap { ArgumentSerializer::Deserialize(argumentString) };
+}
+```
+# Full API Details
+
+```c#
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass Text
+    {
+        Text(String content);
+
+        // The target locale of the XML payload, specified as a BCP-47 language tags.
+        Text SetLanguage(String language);
+
+        // Center the text for incoming call notifications. This value is only used for notifications with with a scenario value of IncomingCall.
+        Text UsesCallScenarioAlign();
+
+        // Displays text at the bottom of your notification, along with your app's identity or the notification's timestamp.
+        // On older versions of Windows that don't support attribution text, the text will simply be displayed as another text element.
+        Text UsesAttributionText();
+
+        // Retrieves the XML content of the Text.
+        String GetXml();
+    };
+
+    runtimeclass ArgumentSerializer
+    {
+        ArgumentSerializer();
+
+        // Adds a key(without value) to the activation arguments that will be returned when the App Notification or its buttons are clicked.
+        ArgumentSerializer AddArgument(String key);
+
+        // Adds a key/value to the activation arguments that will be returned when the App Notification or its buttons are clicked.
+        [default_overload]
+        ArgumentSerializer AddArgument(String key, String value);
+        ArgumentSerializer AddArgument(String key, Int32 value);
+        ArgumentSerializer AddArgument(String key, Double value);
+        ArgumentSerializer AddArgument(String key, Single value);
+        ArgumentSerializer AddArgument(String key, Boolean value);
+
+        // Builds a string from the added arguments to be used in the arguments attribute
+        String Serialize() { get; };
+
```

Comment:
I don't think having `{ get;}`  in the `String Serialize() { get; };` declaration makes much sense. Serialize is a function call (that will do some actual work: it's not a property)